Support checking format for 'gr data.csv'

Ellen99 opened the pull request:
PR has been closed by mistake(sorry for that), I added all the changes in this PR, 

https://github.com/TestingResearchIllinois/idoft/pull/770


Please let me know if you would like me to rename the file 'pr_checker' -> 'checker'.

darko-marinov commented on format_checker/pr_checker.py:
```diff
@@ -182,5 +182,24 @@ def run_checks_pr(log, commit_range):
+    run_checks(filename, data, log, commit_range, checks)
+    check_sort(filename, log)
+
+def run_checks_gr(log, commit_range):
```
Isn't this the same function as `run_checks_pr` but just applied on a different file?

Ellen99 replied:
I didn't add an input parameter for this function because in that case I should also change the code for `run_checks_tic_fic` and `run_checks_tso_iso` functions, all the checkers are called from for loop in `main.py` (lines 21-23).

Should I update them and add fileName parameter?

darko-marinov replied:
Maybe change https://github.com/TestingResearchIllinois/idoft/blob/main/format_checker/main.py#L21 to something like `checks = [lambda: run_checks_pr 'pr-data.csv', lambda: run_checks_pr 'gr-data.csv', run_checks_tic_fic, run_checks_tso_iso]`

Ellen99 replied:
Thanks for that, I'm quite new to python.
Fixed

darko-marinov commented on format_checker/main.py:
```diff
@@ -5,7 +5,7 @@
 from tso_iso_checker import run_checks_tso_iso
 from tic_fic_checker import run_checks_tic_fic
-from pr_checker import run_checks_pr
+from pr_checker import run_checks_pr,run_checks_gr
```
Missing space after comma; are we running some style checker on these files?

Ellen99 replied:
fixed,
I haven't noticed any style checking functions/calls in the repo

darko-marinov replied:
Can you add `flake8`, if familiar with Python? If not, it's not super needed, because these scripts are not maintained by many people.

Ellen99 replied:
A workflow for flake8 is added, there are several style errors in existing python files, I'll resolve them and commit in this PR

Ellen99 replied:
done

darko-marinov commented on format_checker/common_checks.py:
```diff
@@ -92,7 +92,7 @@ def check_sort(filename, log):
     if diff != "":
         log_esp_error(filename, log, "The file is not properly ordered")
-        print("Refer to IDoFT readme for how pr-data.csv should be sorted: https://github.com/TestingResearchIllinois/idoft#to-contribute-a-newly-detected-flaky-test")
+        print("Refer to IDoFT readme for how pr-data.csv, gr-data.csv should be sorted: https://github.com/TestingResearchIllinois/idoft#to-contribute-a-newly-detected-flaky-test")
```
Could this print only the file name being checked (and where the check is failing) rather than printing both file names?

Ellen99 replied:
fixed

darko-marinov reviewed (changes requested):
We're almost done. I have just minor style issues, assuming you ran the new checks? Does the file pass?

Ellen99 commented:
> We're almost done. I have just minor style issues, assuming you ran the new checks? Does the file pass?

Yes the file passed the check.

Ellen99 commented:
I just noticed that `Java CI Maven (auto-update.yml)` workflow has been failing since last year. We are not supporting it anymore, I guess I need to discard my changes from that file.

darko-marinov commented:
Thanks for noticing that `Java CI Maven (auto-update.yml)` workflow is not working. As I said, I forgot that it even exists. Do you want to make it work, or should we disable it?

darko-marinov commented:
Is this PR (getting) ready to be merged, or are you planning to abandon it?

Ellen99 commented:
> Is this PR (getting) ready to be merged, or are you planning to abandon it?

I apologize for the oversight regarding this PR. I got momentarily sidetracked and lost my reminder note to continue working on it. I'll work on it immediately to ensure it is ready for merging as soon as possible.

Ellen99 commented:
> Thanks for noticing that `Java CI Maven (auto-update.yml)` workflow is not working. As I said, I forgot that it even exists. Do you want to make it work, or should we disable it?

I have made necessary changes to make sure auto-update.yml workflow is working properly.

I have updated pom.xml to have the needed versions of dependencies. 
I have also updated the Main.java code to check line contains ",Opened," instead of "Opened" not to treat tests which have "Opened" in their paths as tests that have "Opened" Pull Requests(For Example pr-data.csv has a test named com.alibaba.wasp.master.TestOpenedEntityGroupHandler.testOpenedEntityGroupHandlerOnMasterRestart, this test was resulting in the failure as it contained "Opened" in its name).

darko-marinov commented:
Hopefully this all works fine. @zzjas can do some more checks later if needed.
